siphon/catalog.py

```diff
@@ -342,6 +453,120 @@ def add_access_element_info(self, access_element):
+                outfile.write(infile.read())
+
+    def remote_open(self):
+        """Open the remote dataset for random access.
+
+        Get a file-like object for reading from the remote dataset, providing random access,
+        similar to a local file.
+
+        Returns
+        -------
+        A random access, file-like object
+
+        """
+        return self.access_with_service('HTTPServer')
+
+    def remote_access(self, service=None):
+        """Access the remote dataset.
+
+        Open the remote dataset and get a netCDF4-compatible `Dataset` object providing
+        index-based subsetting capabilities.
+
+        Parameters
+        ----------
+        service : str, optional
+            The name of the service to use for access to the dataset, either
+            'CDMRemote' or 'OPENDAP'. Defaults to 'CDMRemote'.
+
+        Returns
+        -------
+        Dataset
+            Object for netCDF4-like access to the dataset
+
+        """
+        if service is None:
+            service = 'CDMRemote' if 'CDMRemote' in self.access_urls else 'OPENDAP'
+
+        if service not in ('CDMRemote', 'OPENDAP'):
+            raise ValueError(service + ' is not a valid service for remote_access')
+
+        return self.access_with_service(service)
+
+    def subset(self, service=None):
+        """Subset the dataset.
+
+        Open the remote dataset and get a client for talking to ``service``.
+
+        Parameters
+        ----------
+        service : str, optional
+            The name of the service for subsetting the dataset. Defaults to 'NetcdfSubset'.
+
+        Returns
+        -------
+        a client for communicating using ``service``
+
+        """
+        if service is None:
+            service = 'NetcdfSubset'
+
+        if service not in ('NetcdfSubset',):
+            raise ValueError(service + ' is not a valid service for subset')
+
+        return self.access_with_service(service)
+
+    def access_with_service(self, service):
+        """Access the dataset using a particular service.
+
+        Return an Python object capable of communicating with the server using the particular
+        service. For instance, for 'HTTPServer' this is a file-like object capable of
+        HTTP communication; for OPENDAP this is a netCDF4 dataset.
+
+        Parameters
+        ----------
+        service : str
+            The name of the service for accessing the dataset
+
+        Returns
+        -------
+            An instance appropriate for communicating using ``service``.
+
+        """
+        if service == 'CDMRemote':
+            from .cdmr import Dataset as CDMRDataset
+            provider = CDMRDataset
+        elif service == 'OPENDAP':
+            try:
+                from netCDF4 import Dataset as NC4Dataset
+                provider = NC4Dataset
+            except ImportError:
+                raise ValueError('OPENDAP access requires netCDF4-python to be installed.')
```

Comment:
Not sure I agree on that still. They CAN use the service (i.e. it's available on the server), but they need a module installed locally.